client/components/disputed-order-notice/index.js

```diff
@@ -0,0 +1,210 @@
+import moment from 'moment';
+import React, { useEffect } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { dateI18n } from '@wordpress/date';
+import { createInterpolateElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import InlineNotice from 'components/inline-notice';
+import { formatExplicitCurrency } from 'utils/currency';
+import { reasons } from 'wcpay/disputes/strings';
+import { getDetailsURL } from 'wcpay/components/details-link';
+import {
+	isAwaitingResponse,
+	isInquiry,
+	isUnderReview,
+} from 'wcpay/disputes/utils';
+import { useCharge } from 'wcpay/data';
+import wcpayTracks from 'tracks';
+import './style.scss';
+
+const DisputedOrderNoticeHandler = ( { chargeId, onDisableOrderRefund } ) => {
+	const { data: charge } = useCharge( chargeId );
+	const disputeDetailsUrl = getDetailsURL( chargeId, 'transactions' );
+
+	// Disable the refund button if there's an active dispute.
+	useEffect( () => {
+		const { dispute } = charge;
+		if ( ! charge?.dispute ) {
+			return;
+		}
+		// Refunds are only allowed if the dispute is an inquiry or if it's won.
+		const isRefundable =
+			isInquiry( dispute ) || [ 'won' ].includes( dispute.status );
+		if ( ! isRefundable ) {
+			onDisableOrderRefund( dispute.status );
+		}
+	}, [ charge, onDisableOrderRefund ] );
+
+	const { dispute } = charge;
+	if ( ! charge?.dispute ) {
+		return null;
+	}
+
+	// Special case the "under review" notice which is much simpler.
+	// (And return early.)
+	if ( isUnderReview( dispute.status ) ) {
+		return (
+			<DisputeUnderReviewNotice disputeDetailsUrl={ disputeDetailsUrl } />
+		);
+	}
+
+	// Only show the notice if the dispute is awaiting a response.
+	if ( ! isAwaitingResponse( dispute.status ) ) {
+		return null;
+	}
+
+	// Bail if we don't have due_by for whatever reason.
+	if ( ! dispute.evidence_details?.due_by ) {
+		return null;
+	}
+
+	const now = moment();
+	const dueBy = moment.unix( dispute.evidence_details?.due_by );
+
+	// If the dispute is due in the past, don't show notice.
+	if ( ! now.isBefore( dueBy ) ) {
+		return null;
+	}
+
+	return (
+		<DisputeNeedsResponseNotice
```

Comment:
I better understand the role of this component now. It is still quite complex, but the common thing for all the different flavours is that we want the merchant to click `Respond now`.